adr/ADR-8.md

```diff
@@ -122,8 +122,9 @@ type RoKV interface {
 	// History retrieves historic values for a key
 	History(ctx context.Context, key string) ([]Entry, error)
 
-	// WatchBucket watches the entire bucket for changes, all keys and values will be traversed including all historic values
-	WatchBucket(ctx context.Context) (Watch, error)
+	// WatchBucket watches the entire bucket for changes, all keys and values will be traversed including all historic values.
+	// WatchOps can supply modifiers to perform range watches
+	WatchBucket(ctx context.Context, opts ...WatchOpts) (Watch, error)
 
 	// Watch a key for updates, the same Entry might be delivered more than once
 	Watch(ctx context.Context, key string) (Watch, error)
```

Comment:
Instead of watch bucket above why not just allow wc on Watch for key?

Reply:
because a key watch behaves quite differently from a bucket watch - as above bucket watch gives you history. So a scoped bucket watch more describes what is possible with this subset, since JetStream has no feature to give latest per subject.

Reply:
Do we need history though in most cases or do we need to consider adding in the config item from above?

Reply:
With above `LastBySubject` we could drop the history behavior by default but keep it as an opt-in should someone need it, would be quite nice without the history.  Would definitely not mind having that feature.

Reply:
ok may do some R&D..

Reply:
1790e73779501cd0ea9b157333a74435e722814b updates the doc to remove the `WatchBucket()` function entirely and to support ranges on watch, so `Watch(ctx, "") // bucket` or `Watch(ctx, "x.>") // range` or usual `Watch(ctx, "x".y")`

Now both types of watch - bucket, key, range etc - all behave the same thanks to `last_per_subject`